### PR TITLE
Hook up app to API and begin to test without hitting API

### PR DESCRIPTION
**Pull request by riccjohn**
WIP - could use a ton of cleaning up and some tests aren't running correctly
## Description

- Create class to deal with getting auctions from API
- Create class to deal with actually connecting to API
- Create mock client for testing
- Create class to help parse json
- Actually get data from API to be shown in the view

### CODE REVIEW

File: Junk4Dollars-app/API/AuctionsApiService.swift [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+public class AuctionsApiService {
+    let client: APICallable
```
Comment (ijpiantanida) [inline Junk4Dollars-app/API/AuctionsApiService.swift]: You're using 3 different names for the same concept: `client`, `callable` and `query`, which makes it a little bit confusing.
What do you think about using `client` (with slight variations) for everything?

File: Junk4Dollars-app/API/QueryAPI.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public func makeApiCall(endpoint: String, whatToDoWithResponseData: @escaping (Data?, URLResponse?, Error?) -> Void) -> Void {
+        let url = URL(string: endpoint)!
+        let request: URLRequest = URLRequest(url: url)
+
+        let task = URLSession.shared.dataTask(with: request) {data, response, error in
+            if let error = error {
+                print("ERROR: \(error)")
+            }
+
+            guard let data = data else { return }
```
Comment (ijpiantanida) [inline Junk4Dollars-app/API/QueryAPI.swift]: What would happen if the server returns a 500 without a body, or if the connection cannot be established (there's an `error`).

Clients are usually interested in knowing when an API call ends regardless of the result.
Imagine that you start showing a spinner when you start the API call and you need to hide it when it ends.

File: Junk4Dollars-app/API/QueryAPI.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+            whatToDoWithResponseData(data, response, error)
+        }
+
+        task.resume()
+    }
+}
+
+public class MockQueryAPI: APICallable {
```
Comment (ijpiantanida) [inline Junk4Dollars-app/API/QueryAPI.swift]: 👍

File: Junk4Dollars-appTests/ShowAuctions/AuctionListTableViewDataSourceTests.swift [hunk 4]
```diff
@@ -1,23 +0,0 @@
-import XCTest
```
Comment (riccjohn) [inline Junk4Dollars-appTests/ShowAuctions/AuctionListTableViewDataSourceTests.swift]: Bring these tests back and update to test the new behavior

File: Junk4Dollars-appTests/ShowAuctions/AuctionTests.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import Foundation
```
Comment (riccjohn) [inline Junk4Dollars-appTests/ShowAuctions/AuctionTests.swift]: Delete this file

File: Junk4Dollars-appUITests/Junk4Dollars_appUITests.swift [hunk 6]
```diff
@@ -14,10 +14,12 @@ class Junk4Dollars_appUITests: XCTestCase {
         super.tearDown()
     }
 
-    func testTableCellsExist() {
-        let tablesQuery = XCUIApplication().tables
-        XCTAssertTrue(tablesQuery.staticTexts["Throne of Eldraine Booster Box"].exists)
-        XCTAssertTrue(tablesQuery.staticTexts["Oko, Theif of Crowns"].exists)
-        XCTAssertTrue(tablesQuery.staticTexts["MTG M20 Booster Box"].exists)
-    }
+// Can't test this until I can control what the mock api data is
```
Comment (riccjohn) [inline Junk4Dollars-appUITests/Junk4Dollars_appUITests.swift:17]: Let's revisit the UI tests in the future

File: README.md [hunk 7]
```diff
@@ -1,2 +1,10 @@
 # Junk4Dollars iOS app
 [![Build Status](https://travis-ci.com/riccjohn/junk4dollars-app.svg?branch=master)](https://travis-ci.com/riccjohn/junk4dollars-app)
+
+## Architecture Decisions
```
Comment (riccjohn) [inline README.md:4]: Does this need updating?

**Review by ijpiantanida — APPROVED** [review]

**Pull request merged**